tests/workload/network.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package workload
+
+import (
+	"context"
+
+	"github.com/ava-labs/hypersdk/chain"
+)
+
+type TestNetwork interface {
+	Nodes() []TestNode
+	Configuration() TestNetworkConfiguration
+}
+
+type TestNode interface {
+	SubmitTxs(context.Context, []*chain.Transaction) error
+	ConfirmTxs(context.Context, []*chain.Transaction) error
+	GenerateTx(context.Context, []chain.Action, chain.AuthFactory) (*chain.Transaction, error)
+	URI() string
+}
+
+// TestNetworkConfiguration is an interface, implemented by the custom-vm network test framework
+// that allows the test to store information regarding the test network prior to it's invocation, and
+// retrieve the said information during it's execution. It's vital that all implementations of this
+// interface would keep the data stored immutable as it would be shared across multiple threads.
+type TestNetworkConfiguration interface {
+	GenesisBytes() []byte
+	Name() string
+	Parser() chain.Parser
+}
```

Comment:
Can we make this a struct so that VMs can pass in the required fields instead of adding a type that implements this interface?

Reply:
the reason for using an interface here was to allow the casting in the actual test:
```golang
networkConfig := tn.Configuration().(*workload.NetworkConfiguration)
```
just embedding structures would not allow me to do that.
i.e. 
```golang
type TestNetworkConfiguration struct {
	genesisBytes []byte
	name         string
	parser       chain.Parser
}
...
( in the test file )
type NetworkConfiguration struct {
	workload.TestNetworkConfiguration
	keys []ed25519.PrivateKey
}
```
wouldn't let me access the `NetworkConfiguration` by providing a pointer to `workload.TestNetworkConfiguration`. however, if I were to implement an interface, than I would be able to do so.

This would clearly be more important as some VMs would have more setup data that need to be shared with the tests.